src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/util/HttpCacheUtils.java

```diff
@@ -186,60 +190,132 @@ private static boolean responseSatisfiesRequestDirectives(
-                }
+                return true;
             }
-            int ageExpiresInt;
-            if (ageExpires == null)
+            return expires.after(receivedDate);
+        }
+        catch (Exception e)
+        {
+            // Error so just expire it
+            return true;
+        }
+    }
+
+    public static int getAgeExpiresInt(
+            ListFW<HttpHeaderFW> responseHeaders,
+            ReceivedDateHolder receivedDateHolder,
+            Date receivedDate,
+            String ageExpires) throws ParseException
+    {
+        if (ageExpires == null)
+        {
+            String lastModified = getHeader(responseHeaders, LAST_MODIFIED);
+            if (lastModified == null)
             {
-                String lastModified = getHeader(responseHeaders, "last-modified");
-                if (lastModified == null)
-                {
-                    ageExpiresInt = 5000; // default to 5
-                }
-                else
+                return 5000; // default to 5
+            }
+            else
+            {
+                if (!receivedDateHolder.usedLastModifiedHeader)
                 {
                     Date lastModifiedDate = DATE_FORMAT.parse(lastModified);
-                    ageExpiresInt = (int) ((receivedDate.getTime() - lastModifiedDate.getTime()) * (10.0f/100.0f));
+                    return (int) ((receivedDate.getTime() - lastModifiedDate.getTime()) * (10.0f/100.0f));
                 }
+                return 0; // receivedDate was parsed from "last-modified" header
             }
-            else
+        }
+        return Integer.parseInt(ageExpires) * 1000;
+    }
+
+    public static String getAgeExpires(ListFW<HttpHeaderFW> responseHeaders)
+    {
+        String ageExpires = null;
+        String responseCacheControl = HttpHeadersUtil.getHeader(responseHeaders, CACHE_CONTROL);
+        if (responseCacheControl != null)
+        {
+            CacheControlParser parsedCacheControl = new CacheControlParser(responseCacheControl);
+            ageExpires = parsedCacheControl.getValue("s-maxage");
+            if (ageExpires == null)
             {
-                ageExpiresInt = Integer.parseInt(ageExpires) * 1000;
+                ageExpires = parsedCacheControl.getValue("max-age");
             }
-            final Date expires = new Date(System.currentTimeMillis() - ageExpiresInt);
-            return expires.after(receivedDate);
+        }
+        return ageExpires;
+    }
+
+    public static ReceivedDateHolder getReceivedDate(ListFW<HttpHeaderFW> responseHeaders)
+    {
+        ReceivedDateHolder result = new ReceivedDateHolder();
+        String dateHeader = getHeader(responseHeaders, "date");
+        if (dateHeader == null)
+        {
+            dateHeader = getHeader(responseHeaders, LAST_MODIFIED);
+            result.usedLastModifiedHeader = true;
+        }
+        if (dateHeader == null)
+        {
+            // invalid response, so say it is expired
```

Comment:
no need for NULL check if you just catch all exceptions and do EPOCH in all exception cases